### PR TITLE
Vector pop push

### PR DESCRIPTION
**Pull request by thomasmulvaney**

### CODE REVIEW

File: pixie/vm/persistent_vector.py [hunk 1]
```diff
@@ -350,11 +350,7 @@ def nth(self, i, not_found=nil):
 
     def pop(self):
         self.ensure_editable()
-        affirm(self._cnt != 0, u"Can't pop and empty vector")
-
-        if self._cnt == 0:
-            self._cnt = 0
-            return self
```
Comment (heyLu) [inline pixie/vm/persistent_vector.py]: @halgari Do you know what this was supposed to do?

**Comment by heyLu** [comment]
Is there a clojure equivalent for `push` or is it just `conj`?

**Comment by thomasmulvaney** [comment]
In clojure there is only `conj`. Interestingly, clojures `IPersistentStack` protocol has `peek` and `pop`, perhaps pixie's `IStack` should have `-peek` too?

**Comment by arrdem** [comment]
Adding `-peek` seems entirely reasonable given that `peek`, `push` and `pop` traditionally define stack types although one could argue that `first` of a stack should have the same effect.

**Comment by halgari** [comment]
for vectors, push = conj, but I suppose it could be different for other data structures, so I'm fine with having both for now.

**Pull request merged**